### PR TITLE
smtp/mime: Restrict file name lengths

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #4526 

This commit places restrictions on the length of the file name specified
in attachments (`name=` or `filename=`) to `NAME_MAX`. Names exceeding
these limits will be truncated and processing will continue with the
truncated name.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [3409](https://redmine.openinfosecfoundation.org/issues/3265)

Describe changes:
- Address review comments
- Add test case.

### CODE REVIEW

File: src/app-layer-smtp.c [hunk 1]
```diff
@@ -558,7 +560,7 @@ int SMTPProcessDataChunk(const uint8_t *chunk, uint32_t len,
  *
  * \param state The smtp state.
  *
- * \retval  0 On suceess.
+ * \retval  0 On succeess.
```
Comment (victorjulien) [inline src/app-layer-smtp.c]: Not sure if this improves things much :-P

File: src/util-decode-mime.c [hunk 2]
```diff
@@ -3035,6 +3080,65 @@ static int MimeIsIpv6HostTest01(void)
...
+    MimeDecGetConfig()->decode_base64 = 1;
+    MimeDecGetConfig()->decode_quoted_printable = 1;
+    MimeDecGetConfig()->extract_urls = 1;
+
+    /* Init parser */
+    MimeDecParseState *state = MimeDecInitParser(&line_count,
+            TestDataChunkCallback);
+
+    const char *str = "From: Sender1";
+    ret |= MimeDecParseLine((uint8_t *)str, strlen(str), 1, state);
```
Comment (victorjulien) [inline src/util-decode-mime.c]: Can you wrap each individual call in a `FAIL_IF_NOT`? E.g.
```
FAIL_IF_NOT(MimeDecParseLine((uint8_t *)str, strlen(str), 1, state) == MIME_DEC_OK);
```

File: src/app-layer-smtp.c (hunk 1 above)
Comment (jlucovsky) [reply src/app-layer-smtp.c]: Yikes.

**Comment by jlucovsky** [comment]
Continued in #4556

**Pull request closed without merging**